numpy/lib/tests/test_index_tricks.py

```diff
@@ -273,6 +291,12 @@ def test_accepts_npcomplexfloating(self):
             mgrid[0.1:0.3:3j], mgrid[0.1:0.3:np.complex64(3j)]
         )
 
+        # Related to #16945
+        assert_array_equal(
```

Comment:
As far as I remember, `assert_array_equal` does not check the return type, only the value

Reply:
Yup, you're right.
I added a few explicit dtype checks in https://github.com/numpy/numpy/pull/16971/commits/3a49c7c750fddc2093700d8d844155be86c0f302.